Makefile

```diff
@@ -55,6 +55,24 @@ $(OUT)/emulate.o: CFLAGS += -fno-gcse -fno-crossjumping
 endif
 endif
 
+MINI_GDBSTUB :=
+ENABLE_GDBSTUB ?= 1
+ifeq ("$(ENABLE_GDBSTUB)", "1")
+CONFIG_SOCKET_ADDR = 127.0.0.1:1234
+CFLAGS += -D ENABLE_GDBSTUB -D'SOCKET_ADDR="$(CONFIG_SOCKET_ADDR)"'
+LDFLAGS += -Wl,-rpath="$(CURDIR)/mini-gdbstub" -L./mini-gdbstub -lgdbstub
+MINI_GDBSTUB += mini-gdbstub/libgdbstub.so
+$(MINI_GDBSTUB):
+	git submodule update --init mini-gdbstub
+	$(MAKE) -C mini-gdbstub
```

Comment:
Can you build the intermediate files such as `.o` in the `build` directory? That is, [out-of-tree build](https://unix.stackexchange.com/questions/208638/linux-kernel-meaning-of-source-tree-in-tree-and-out-of-tree).